include/mbedtls/net_sockets.h

```diff
@@ -123,6 +123,24 @@ void mbedtls_net_init( mbedtls_net_context *ctx );
  */
 int mbedtls_net_connect( mbedtls_net_context *ctx, const char *host, const char *port, int proto );
 
+/**
+ * \brief          Initiate a connection with host:port in the given protocol with a timeout
```

Comment:
Please create a changelog entry file in `Changelog.d` announcing the new function (see “Features” entries in `ChangeLog` for similar examples).